source/_posts/2020-09-16-supervisor-joins-the-party.markdown

```diff
@@ -0,0 +1,95 @@
+
+Add-on developers will be able to activate additional functionality for their add-on with the new “watchdog” add-on option. This enables application-level monitoring and allows the Supervisor to check if the add-on is behaving correctly using the ways specified by the developer.
+
+Not all add-ons have this feature and an add-on can still get themselves into problems that they can’t recover from, causing the application not to work as expected and crash. For this purpose we have introduced a new watchdog toggle for advanced users. The functionality is similar to Docker’s health checks but operate outside of the container, making it a more robust option.
+
+![watchdog graphic](/images/blog/2020-09-16-supervisor-joins-the-party/watchdog.png)
+
+When the watchdog is enabled for an add-on, the Supervisor will automatically restart the add-on if it stops, regardless of the reason (crash/manual stop). The Watchdog does not know if you’re testing an add-on or playing around with different options, and so it might restart when you don’t need it. You should only enable the watchdog after you are finished setting it up and want to make sure it’s running 24/7.
+
+<blockquote>
+   "Never stop never stopping." - Conner4Real
+</blockquote>
+
+## Network Manager
+
+The most requested feature is here! You can now manage the network settings from the Supervisor interface. Previously you would have to fiddle with “nmcli” or go through the tedious act of importing configuration files from USB-sticks just to set a static IP address for your Home Assistant installation. With the introduction of the Supervisor’s Network Manager this can now be handled from the Supervisor panel in your Home Assistant UI. You will find this under the system tab.
+
+![dialog graphic](/images/blog/2020-09-16-supervisor-joins-the-party/dialog.png)
+
+This is just the start to make advanced host management more accessible and it only touches a minimal part of what we will be able to do with it. Support for Wi-Fi and Bluetooth will follow later as part of our usability improvements.
+
+<blockquote>
+   "OMG. Finally." - @cogneato
+</blockquote>
+
+## Observer Plugin
+
+The Supervisor provides several other services using “plugins”. Plugins are microservices that add features to help run Home Assistant + add-ons and manage your system. The latest plugin added is the [Observer plugin][observer]. The observer keeps an eye on the Supervisor. It provides a diagnostics portal on port `4357`(HELP). In case you ever lose access to the Supervisor, you can go to th eObserver Portal to get the Supervisor status and logs and diagnose what’s wrong (and share it with us). The benefit of having this as a web portal is that you no longer need to attach a monitor and keyboard to your device or know the Linux commands to get the information out.
```

Comment:
```suggestion
The Supervisor provides several other services using “plugins”. Plugins are microservices that add features to help run Home Assistant + add-ons and manage your system. The latest plugin added is the [Observer plugin][observer]. The observer keeps an eye on the Supervisor. It provides a diagnostics portal on port `4357`(HELP). In case you ever lose access to the Supervisor, you can go to the Observer Portal to get the Supervisor status and logs and diagnose what’s wrong (and share it with us). The benefit of having this as a web portal is that you no longer need to attach a monitor and keyboard to your device or know the Linux commands to get the information out.
```